src/main/java/com/aws/greengrass/componentmanager/builtins/GreengrassRepositoryDownloader.java

```diff
@@ -78,14 +87,18 @@ public Long getDownloadSize() throws PackageDownloadException, InterruptedExcept
     }
 
     private Long getDownloadSizeWithoutRetry() throws InterruptedException, PackageDownloadException, IOException {
-        URL url = getArtifactDownloadURL(identifier, artifact.getArtifactUri().getSchemeSpecificPart());
+        String url = getArtifactDownloadURL(identifier, artifact.getArtifactUri().getSchemeSpecificPart());
 
-        HttpURLConnection httpConn = null;
-        try {
-            httpConn = connect(url);
-            int responseCode = httpConn.getResponseCode();
+        try (SdkHttpClient client = getSdkHttpClient()) {
+            HttpExecuteRequest executeRequest = HttpExecuteRequest.builder()
```

Comment:
double check that none of the other objects that it makes are closable such as a response, response stream, etc.